pytensor/link/numba/dispatch/basic.py

```diff
@@ -744,3 +745,22 @@ def ifelse(cond, *args):
             return res[0]
 
     return ifelse
+
+
+@numba_funcify.register(Nonzero)
+def numba_funcify_Nonzero(op, node, **kwargs):
+    a = node.inputs[0]
+
+    if a.ndim == 0:
+        raise ValueError("Nonzero only supports non-scalar arrays.")
```

Comment:
This input validation is done by the `nonzero` Op itself, there's no need to repeat it here.